# Reads alignment

<figure align = "center">
  <img src="https://github.com/sujin9819/MetaInsight/blob/main/SOP/MetaProteomic/img/P_6_1.png?raw=true" style="width:90%">
  <figcaption><b> </b></figcaption>  
</figure>

We would like to introduce the "reference-guided analysis" method, which is based on mapping RNA-seq reads to the sequences of de novo assembled contigs and metagenome-assembled genomes (MAGs) obtained from metagenome analysis. 

<figure align = "center">
  <img src="https://github.com/sujin9819/MetaInsight/blob/main/SOP/MetaProteomic/img/P_6_2.png?raw=true" style="width:90%">
  <figcaption><b> </b></figcaption>  
</figure>

In this approach, after pre-processing, where fastq files are processed to enhance sequence quality, alignment is performed by mapping the reads to the reference sequences*. The STAR program is utilized to create the reference file and perform the mapping of reads to the generated reference sequences.  
- **Types of Reference Sequences**
1.Final.contig.fa obtained from metagenome *de novo* assembly pipeline analysis.
2. MAG.fa obtained from metagenome MAG binning results.

```bash
#building STAR index
$ STAR --runMode genomeGenerate --genomeSAindexNbases 10 --runThreadN 8 --genomeDir ./index/sample --genomeFastaFiles sample_MAG.fa 
#mapping
$ STAR --genomeDir ./index/sample --runThreadN 8 --readFilesIn ./1.Trim/sample_kneaddata_paired_1.fastq ./1.Trim/sample_kneaddata_paired_2.fastq --outFileNamePrefix ./2.Align/sample_
```
By utilizing the STAR `runMode` command, index files for STAR running are generated from the MAG.fa file of the metagenome. These created index files serve as a basis for mapping reads to reference sequences, resulting in a .sam file where information about gene counts is assigned to each gene.
The STAR alignment results are not output separately but are stored in the final.out file. The results of read alignment can be verified using the Log.final.out file, which provides information about the total unique reads, multi-mapping reads, and unmapped reads. Due to the large size of the STAR-generated .sam files, they are converted into smaller-sized binary format .bam files.  
To align the mapping information after read mapping, the samtools' `view` and `sort` functions are utilized to transform the .sam files into .bam files. 

```bash
#install samtools
$ tar -xvf samtools.zip
$ cd samtools
$ ./configure -prefix==/where/to/install
$ make
$ make install
#sam file sorting and indexing
$ samtools view -b -F 4 sample_Aligned.out.sam | samtools sort - > sample.bam
$ samtools index sample.bam
```

Once these .bam files are generated, you can visualize the alignment results using the IGV program. To examine the alignment results with the IGV program, you will need the MAG.fa obtained from metagenome de novo assembly or MAG results. If you wish to display annotated coding sequences (CDS) information within the MAG.fa sequences, you can import an annotation-generated .gff file* as well.
Additionally, you will require the .bam files containing coverage value information resulting from the application of samtools and the .bam.bai file generated by `samtools index`. Once all these files are imported, the visualization will display information on CDS in the reference sequence, the location of mapping of metatranscriptomic sequence reads, and the count of metatranscriptomic reads mapped at each position.  
*Note: The creation of a .gff (General Feature Format) file for MAG.fa can be referred to in the metagenome SOP.

<figure align = "center">
  <img src="https://github.com/sujin9819/MetaInsight/blob/main/SOP/MetaProteomic/img/P_6_3.png?raw=true" style="width:90%">
  <figcaption><b>Visualization of alignment result using IGV program</b></figcaption>  
</figure>

```bash
# download prodigal from https://github.com/hyattpd/Prodigal/wiki/installation
# install prodigal
$ cd prodigal/
$ make install
# run prodigal
$ prodigal -i sample_MAG.fa -p meta -a sample_MAG.prodigal.faa -d sample_MAG.prodigal.fna -f gff -o sample_MAG.prodigal.gff 
```
Using Bedtools, you can obtain coverage (count) information for each CDS (coding sequence) in functional annotations. Before applying Bedtools, it is advisable to simplify the .gff entries to include only the essential information for Bedtools usage.
You can utilize the ParseGFFonlyCDS.R R script to parse the comprehensive .gff file obtained from functional annotation, extracting only the CDS information, CDS location, and locus tag information to create a simplified .gff file.

```bash
#make CDS only gff file with python-script
$ wget https://raw.githubusercontent.com/sujin9819/ngs/main/R_scripts/ParseGFFonlyCDS.R 
$ Rscript ParseGFFonlyCDS.R -i sample_MAG.prodigal.gff -o MAG.CDS.gff
```

The Bedtools' `bamtobed` command converts the bam file into a bed file, and the `bedtools coverage` command is used to derive count values based on entries in the gff file from the bed file. By matching the location information of CDS entries in the gff file with the locations where sequence reads from the bed file are mapped, you can calculate the count values for each CDS.
The resulting bed file includes information about CDS locations, counts of sequence reads mapped to each CDS, and information about the strand direction, in the case of genes (as shown in the example below, denoted as sample.cov).

```bash
#install bedtools
$ tar -zxvf bedtools-2.29.1.tar.gz
$ cd bedtools2
#alignment
$ bedtools bamtobed –i sample.bam > sample.bed
$ bedtools coverage –a MAG.CDS.gff –b sample.bed –s > sample.cov


<figure align = "center">
  <img src="https://github.com/sujin9819/MetaInsight/blob/main/SOP/MetaProteomic/img/P_6_4.png?raw=true" style="width:90%">
  <figcaption><b>Example bed file generated as a result of read alignment</b></figcaption>  
</figure>
