# Standard Operating Procedure for Next-Generation Sequencing (NGS)-Based Metagenome Analysis

## Introduction to metagenome

<figure align = "center">
  <img src="https://github.com/sujin9819/MetaInsight/blob/main/SOP/MetaGenomic/img/G_0_1.png?raw=true" style="width:90%">
  <figcaption><b>Metagenomics overview</b></figcaption>  
</figure>

A metagenome is the pool of genetic material present in various environmental samples (animal, plant, marine, etc.).
Metagenomic analysis extracts and analyzes the genetic information of all organisms present in a given environmental ecosystem.
Before the rise of metagenomics, microbial genomes could only be analyzed from culturable microorganisms. However, since only a small fraction of the microbes in a given environment are culturable, this approach results in biased information centered on culturable microbes.
Metagenomic analyses, on the other hand, analyze the entire genome of an environment, providing a more accurate picture of a given ecosystem.

Advancements in sequencing technology have made it easier for researchers to access whole metagenome shotgun sequencing, which sequences billions of nucleic acid fragments at once, beyond amplicon sequencing, which involves targeted amplification and sequencing of specific parts of the genome such as 16S ribosomal DNA, and the demand for research has increased.
For amplicon sequencing data, there are standard analysis protocols and platforms such as QIIME2, but for WMS analysis, there is no standard gold-standard analysis SOP.
In this SOP, we will describe the methods and flow for analyzing whole genome sequencing (WGS) data, focusing on the most commonly used tools among various analysis programs.
